tests/test_processing_common.py

```diff
@@ -0,0 +1,169 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import shutil
+
+import numpy as np
+import pytest
+from PIL import Image
+
+from transformers.testing_utils import require_torch
+
+
+@require_torch
+class ProcessorTesterMixin:
+    tokenizer_class = None
+    fast_tokenizer_class = None
+    image_processor_class = None
+    processor_class = None
+
+    def get_tokenizer(self, **kwargs):
+        return self.tokenizer_class.from_pretrained(self.tmpdirname, **kwargs)
+
+    def get_fast_tokenizer(self, **kwargs):
+        return self.fast_tokenizer_class.from_pretrained(self.tmpdirname, **kwargs)
+
+    def get_image_processor(self, **kwargs):
+        return self.image_processor_class.from_pretrained(self.tmpdirname, **kwargs)
+
+    def tearDown(self):
+        shutil.rmtree(self.tmpdirname)
+
+    def prepare_image_inputs(self):
+        """This function prepares a list of PIL images, or a list of numpy arrays if one specifies numpify=True,
+        or a list of PyTorch tensors if one specifies torchify=True.
+        """
+
+        image_inputs = [np.random.randint(255, size=(3, 30, 400), dtype=np.uint8)]
+
+        image_inputs = [Image.fromarray(np.moveaxis(x, 0, -1)) for x in image_inputs]
+
+        return image_inputs
+
+    def test_save_load_pretrained_default(self):
+        tokenizer_slow = self.get_tokenizer()
+        tokenizer_fast = self.get_fast_tokenizer()
+        image_processor = self.get_image_processor()
```

Comment:
Could use `get_components` (suggested in the above comment) to get all the components to instantiate a processor.

The test could be rewritten to handle all the possible component types rather than just tokenizer+image as below.